pkg/stub/handler.go

```diff
@@ -61,7 +61,7 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 			logrus.Errorf("failed to update %v", o)
 			return err
 		}
-	case *appsv1.Deployment:
+	default:
```

Comment:
To accept a larger class of Pod types without replicating code, I added this change. 

However, this would cause conflicts with sdk.List here - https://github.com/jaegertracing/jaeger-operator/pull/78/files#diff-2d7ceed365f1ea47bba386dac808cb07R72 - where we need explicit type checking.

Reply:
Instead of accepting a more generic pod type there, you could just list all the types that should be included in this branch, like:

```golang
case *appsv1.Deployment:
case *appsv1.StatefulSet:
  ....
```

Reply:
Oh. Did you mean a switch-case fall-through? We would still have to write code for type checking the `sdk.Object` in the inject package. I've made some changes, do let me know if OK. @jpkrohling

Reply:
Something like that, but not quite: you can keep the two cases collapsed into one, like the example in previous comment. With more context, the example would be like this:

```golang
func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
	switch o := event.Object.(type) {
	case *v1alpha1.Jaeger:
            ....
	case *appsv1.Deployment:
	case *appsv1.StatefulSet:
            if inject.Needed(o) {...}
	}
	return nil
```

The `inject.Needed` would have to be changed. Perhaps instead of accepting a `sdk.Object` (or `runtime.Object`, it could accept two parameters: a map[string]string with the annotations and a `PodTemplateSpec`. The same would probably work with the other functions, like `inject.Select` and `inject.Sidecar`. Adding another supported object type (DaemonSet) would then just be a matter of adding it to the switch case in the handler.